### PR TITLE
feat: [PIE-6047]: add react-query integration

### PR DESCRIPTION
**Pull request by vkbansal-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/services/cd-ng-rq/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+export type {
```
Comment (AshwinBhatkal-Harness) [inline src/services/cd-ng-rq/index.ts:1]: Is this file autogenerated?

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

**Review by satyam-harness — DISMISSED** [review]

File: src/services/cd-ng-rq/index.ts (hunk 1 above)
Comment (vkbansal-harness) [reply src/services/cd-ng-rq/index.ts:1]: yes

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/useGetServicesData.ts [hunk 2]
```diff
@@ -40,6 +35,8 @@ export interface UseGetServicesDataReturn {
   refetchListData(): void
   prependServiceToServiceList(newServiceInfo: ServiceYaml): void
 }
+// react-query staleTime
+const STALE_TIME = 60 * 1000 * 15
```
Comment (udayvunnam) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/useGetServicesData.ts:39]: Good to have a generic stale time at the root level.

File: configs/oats.config.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      },
+      genOnlyUsed: true,
+      plugins: [
+        reactQueryPlugin({
+          customFetcher: 'services/fetcher',
+          allowedOperationIds: ['getServicesYamlAndRuntimeInputs', 'getServiceAccessList'],
+          overrides: {
+            getServicesYamlAndRuntimeInputs: {
+              useQuery: true
+            }
```
Comment (udayvunnam) [inline configs/oats.config.ts:36]: nice!

**Review by udayvunnam — APPROVED** [review]

**Review by kapilkumawat86 — DISMISSED** [review]

**Review by ishantmehtaharness — APPROVED** [review]

File: __mocks__/@tanstack/react-query.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports.useQuery = () => {
+  fail('Please mock your API calls')
```
[the remark this answers is not in the document]
Comment (vkbansal-harness) [reply __mocks__/@tanstack/react-query.js:2]: Will do in follow up PR

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/framework/app/App.tsx [hunk 5]
```diff
@@ -74,6 +76,7 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
   // if user lands on /, they'll first get redirected to a path with accountId
   const { accountId } = useParams<AccountPathProps>()
   const { forceLogout } = useLogout()
+  const token = SessionToken.getToken()
```
[the remark this answers is not in the document]
Comment (vkbansal-harness) [reply src/framework/app/App.tsx]: reverted the change

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**